x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/AllocationFilteringIntegTests.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    public void testReplacesExcludeFilter() throws InterruptedException {
+        runTest(INDEX_ROUTING_EXCLUDE_GROUP_SETTING, false, INDEX_ROUTING_EXCLUDE_GROUP_SETTING, false);
+    }
+
+    public void testReplacesIncludeFilterWithExcludeFilter() throws InterruptedException {
+        runTest(INDEX_ROUTING_INCLUDE_GROUP_SETTING, true, INDEX_ROUTING_EXCLUDE_GROUP_SETTING, false);
+    }
+
+    /**
+     * Starts two nodes, allocates the original index on the first node but then excludes that node and verifies that the index can still be
+     * mounted and allocated to the other node.
+     *
+     * @param indexSetting           an allocation filter setting to apply to the original index
+     * @param mountSetting           an (optional) allocation filter setting to apply at mount time
+     * @param indexSettingIsPositive whether {@code indexSetting} is positive (i.e. include/require) or negative (i.e. exclude)
+     * @param mountSettingIsPositive whether {@code mountSetting} is positive (i.e. include/require) or negative (i.e. exclude)
+     */
+    private void runTest(
+        Setting.AffixSetting<String> indexSetting,
+        boolean indexSettingIsPositive,
+        @Nullable Setting.AffixSetting<String> mountSetting,
+        boolean mountSettingIsPositive
+    ) throws InterruptedException {
+        final List<String> nodes = internalCluster().startNodes(2);
+
+        // apply an index setting to restrict the original index to node 0
+        final Settings.Builder indexSettings = Settings.builder()
+            .put(indexSetting.getConcreteSettingForNamespace("_name").getKey(), nodes.get(indexSettingIsPositive ? 0 : 1));
+
+        final Settings.Builder mountSettings = Settings.builder();
+        if (mountSetting != null) {
+            // apply an index setting to restrict the mounted index to node 1
+            mountSettings.put(mountSetting.getConcreteSettingForNamespace("_name").getKey(), nodes.get(mountSettingIsPositive ? 1 : 0));
+        }
+
+        final MountSearchableSnapshotRequest mountRequest = prepareMountRequest(indexSettings, mountSettings);
+
+        // block allocation to node 0 at the cluster level
+        assertAcked(
+            client().admin()
+                .cluster()
+                .prepareUpdateSettings()
+                .setPersistentSettings(
+                    Settings.builder()
+                        .put(CLUSTER_ROUTING_EXCLUDE_GROUP_SETTING.getConcreteSettingForNamespace("_name").getKey(), nodes.get(0))
+                )
+        );
+
+        // mount snapshot and verify it is allocated as expected
+        final RestoreSnapshotResponse restoreSnapshotResponse = client().execute(MountSearchableSnapshotAction.INSTANCE, mountRequest)
+            .actionGet();
+        assertThat(restoreSnapshotResponse.getRestoreInfo().failedShards(), equalTo(0));
+        ensureGreen(mountRequest.mountedIndexName());
```

Comment:
Would be nice to also check that the setting is not actually set (when it is different from `mountSetting`).

Reply:
Ok, done in 1bcdbc2fbef.